src/OpenLoco/src/Objects/ObjectIndex.cpp

```diff
@@ -223,19 +224,19 @@ namespace OpenLoco::ObjectManager
         _installedObjectCount++;
 
         _isPartialLoaded = true;
-        _50D158 = _112A17F;
-        loadTemporaryObject(objHeader);
-        _50D158 = reinterpret_cast<std::byte*>(-1);
+        _dependentObjectsVector = _dependentObjectVectorData;
+        const bool tempLoadFailed = loadTemporaryObject(objHeader);
+        _dependentObjectsVector = reinterpret_cast<std::byte*>(-1);
         _isPartialLoaded = false;
-        if (_installedObjectCount == 0)
```

Comment:
Today I learned that an x86 `dec` operation does not set the carry flag so this was meant to be the result of `loadTemporaryObject`